torch/csrc/stub.c

```diff
@@ -9,7 +9,11 @@ extern "C"
 __attribute__((visibility("default"))) PyObject* PyInit__C(void);
 #endif
 
+extern void global_kineto_init();
```

Comment:
I think we might need to move this prototype to a header for when stub.c is not compiled with kineto_client_interface

Reply:
Yes, this prototype can be moved to a header file. By moving this prototype to a header, do you mean some macros should be added to determine whether `stub.c` is compiled with `kineto_client_interface`?  Here's a simple example: 

```
#ifndef TORCH_KINETO_H
#define TORCH_KINETO_H

#ifndef USE_KINETO 
// stub.c is compiled with `kineto_client_interface`, `torch._C` can find symbol `global_kineto_init`. 
void global_kineto_init();
#else
// stub.c is not compiled with `kineto_client_interface`,  `PyMODINIT_FUNC PyInit__C(void)` calls a empty function. 
inline void global_kineto_init(){}
#endif
#endif // TORCH_KINETO_H
```

Reply:
I might need to look at the compile paths but I just see this PR failing a lot of builds if you look below

Reply:
Based on the failures it is failing because of missing prototype in kineto_client_interface so we need to have a header that this file and stub.c includes to make the builds compile

Reply:
Sorry, I was out of office last week. I'd check those builds and try to fix it today.

Reply:
I created a new header file `kineto_client_interface.h`, and declare `global_kineto_init` inside this header file. Then,  I moved the call to `global_kineto_init` into `torch/csrc/Module.cpp:initModule` instead of `stub.c` as `torch/csrc/Module.cpp` has already included header files under `torch/csrc/profiler`, that is `#include <torch/csrc/profiler/combined_traceback.h>`.